src/library/scala/collection/immutable/VectorMap.scala

```diff
@@ -64,7 +64,13 @@ final class VectorMap[K, +V] private[immutable] (
   def remove(key: K): VectorMap[K, V] = {
     underlying.get(key) match {
       case Some((index, _)) =>
-        new VectorMap(fields.patch(index, Nil, 1), underlying - key)
+        val finalUnderlying = (underlying - key).map{ case (k, (currentIndex,v)) =>
```

Comment:
For a temporary fix for correctness, this looks reasonable, but overall this doesn't look like a good encoding.  It's better to adopt an amortized strategy where you swap in "deleted" elements with `updated` (`O(log n)`) and eventually clear them out (e.g. if you do when there are `n/log n` dead elements, then you have an `O(n)` cleanup after `O(n)` work, for an overall complexity of `O(n log n)` for removing `n` elements).  If a cleverer strategy like this is adopted, it can prevent the need for an additional RBTree-backed order-preserving map of this type.

Reply:
> For a temporary fix for correctness, this looks reasonable, but overall this doesn't look like a good encoding. It's better to adopt an amortized strategy where you swap in "deleted" elements with updated (O(log n)) and eventually clear them out (e.g. if you do when there are n/log n dead elements, then you have an O(n) cleanup after O(n) work, for an overall complexity of O(n log n) for removing n elements). If a cleverer strategy like this is adopted, it can prevent the need for an additional RBTree-backed order-preserving map of this type.

I have a similar strategy which is that we store a list of all of the elements that we have removed and we use this list to create an adjustment so that we know the correct index. Likewise we have to cleanup after the list gets large enough. Will post a PR later to see what you think